more_itertools/more.py

```diff
@@ -1411,6 +1413,41 @@ def stagger(iterable, offsets=(-1, 0, 1), longest=False, fillvalue=None):
     )
 
 
+class UnequalIterablesException(Exception):
+    def __init__(self, idxs, names=None):
+        if names is None:
+            depleted = "Indices: {}".format(
+                ", ".join(str(idx) for idx in idxs)
+            )
+        else:
+            depleted = "Columns: {}".format(
+                ", ".join(names[idx] for idx in idxs)
+            )
+        super().__init__(
+            "Iterables have different lengths."
+            "{} depleted before rest.".format(depleted)
+        )
+
+
+def zip_equal(*iterables, names=None):
+    """``zip`` the input *iterables* together, but throw an
+    ``UnequalIterablesException`` if any of the *iterables* terminate before
+    the others. The exception will give the indices which terminated too early.
+    Alternatively *names* can be passed and it will give the name corresponding
+    to the early terminating indices.
+
+    """
+    sentinel = object()
```

Comment:
We already have `_marker` in this module; we can use that instead of a new sentinel.